### PR TITLE
ui: refactor model updating and rendering into ModelRenderer class

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

**Comment by maxime-desroches** [comment]
trigger-jenkins

**Comment by maxime-desroches** [comment]
Can you split this PR into 2 PRs:

1: No logical change, renaming or optimization (or at least try to keep it to an absolute minimum). Just move the old logic to the new ModelRenderer class.

2: Apply any cleanup/optimization/refactor/renaming you want.

File: selfdrive/ui/qt/onroad/model.cc [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  for (int i = 0; i <= max_idx; ++i) {
+    float x = x_coords[i], y = y_coords[i], z = z_coords[i];
+    // highly negative x positions  are drawn above the frame and cause flickering, clip to zy plane of camera
+    if (x < 0) continue;
+
+    const QPointF left = mapToScreen(x, y - y_off, z + z_off);
+    const QPointF right = mapToScreen(x, y + y_off, z + z_off);
+    if (clip_region.contains(left) && clip_region.contains(right)) {
+      // For wider lines the drawn polygon will "invert" when going over a hill and cause artifacts
+      if (!allow_invert && !pvd->empty() && left.y() > pvd->back().y()) {
```
Comment (maxime-desroches) [inline selfdrive/ui/qt/onroad/model.cc:187]: Changes like this  (`pvd->size()` to `!pvd->empty()`) should go in a different PR since they are hard to spot and review when so many things are moved around.

**Comment by deanlee** [comment]
submitted [PR #33702](https://github.com/commaai/openpilot/pull/33702), which includes minimal logic changes
